### PR TITLE
Completely replace Vundle with vim-plug

### PR DESCRIPTION
**Pull request by SimpleX91**
Related to #81
It's necessary to change all *Plugin* to *Plug* in *bundles.vim*, so I decided to (temporary at least) change source from *bundles.vim* to *plug.vim* to avoid errors.
It can be done automatically via **install.sh** or manually by user (preferred, I think).

### CODE REVIEW

File: .vimrc [hunk 1]
```diff
@@ -44,64 +44,61 @@ nnoremap Q <nop>
...
 " Support bundles
-Plugin 'jgdavey/tslime.vim'
-Plugin 'Shougo/vimproc.vim'
-Plugin 'ervandew/supertab'
-Plugin 'scrooloose/syntastic'
-Plugin 'moll/vim-bbye'
-Plugin 'nathanaelkane/vim-indent-guides'
-Plugin 'vim-scripts/gitignore'
+Plug 'jgdavey/tslime.vim'
+Plug 'Shougo/vimproc.vim', { 'do': 'make' }
```
Comment (begriffs) [inline .vimrc]: Well this is nice that you can have it do extra tasks like that!

**Comment by begriffs** [comment]
Thanks for sending me this PR.

I'm running into a problem on my machine when I try to run the installer:
```
./install.sh: line 92: 49149 Segmentation fault: 11  vim -T dumb -E -u $endpath/.vimrc +PlugUpgrade! +PlugInstall! +PlugClean! +qall
```

I adjusted the commands to `+PlugUpgrade +PlugClean! +qall` because Upgrade also does an Install. Same segfault though. I'm on a mac with vim version

```
$ vim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Oct 29 2014 15:26:57)
MacOS X (unix) version
Included patches: 1-488
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl             +farsi           +mouse_netterm   +syntax
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
-balloon_eval    +float           +mouse_urxvt     -tag_any_white
-browse          +folding         +mouse_xterm     -tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         -gettext         -mzscheme        +textobjects
-clientserver    -hangul_input    +netbeans_intg   +title
+clipboard       +iconv           +path_extra      -toolbar
+cmdline_compl   +insert_expand   +perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python          +viminfo
+cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con      -lua             +rightleft       +windows
+diff            +menu            +ruby            +writebackup
+digraphs        +mksession       +scrollbind      -X11
-dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     -xim
+emacs_tags      -mouseshape      -sniff           -xsmp
+eval            +mouse_dec       +startuptime     -xterm_clipboard
+ex_extra        -mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    -xpm
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: /usr/bin/clang -c -I. -Iproto -DHAVE_CONFIG_H   -F/usr/local/Frameworks -DMACOS_X_UNIX  -Os -w -pipe -march=native -mmacosx-version-min=10.9 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: /usr/bin/clang   -L. -L/Users/travis/.sm/pkg/active/lib -fPIC -Bstatic -fstack-protector -L/usr/local/lib -F/usr/local/Frameworks -Wl,-headerpad_max_install_names -o vim        -lm  -lncurses -liconv -framework Cocoa   -fstack-protector -L/usr/local/lib  -L/System/Library/Perl/5.16/darwin-thread-multi-2level/CORE -lperl -framework Python   -lruby-static -framework CoreFoundation -lobjc -L/Users/j/.rvm/rubies/ruby-2.1.2/lib
```

Does the installer work OK on your machine?

**Comment by SimpleX91** [comment]
Yes, it works without issues here, Fedora Workstation 23.
PlugUpgrade does not run install, it upgrades vim-plug only.

Run the command like this:
"""vim -T dumb -E -u ~/.vimrc +PlugUpgrade! +PlugInstall! +PlugClean! +qall"""
and look what it prints. If it fails on make, then it's vimproc fails to compile.

Try clean install (remove bundle directory completely).

**Comment by begriffs** [comment]
OK, I'll try this later today or tomorrow and let you know.

**Comment by SimpleX91** [comment]
Hey there! Did you make it work?

**Comment by SimpleX91** [comment]
Cleaned up (rebased) branch to only contain vim-plug related commits.

**Comment by begriffs** [comment]
Pulled latest changes on this branch and ran the command you suggested. It always says

```
Vim: Caught deadly signal SEGV
Vim: Finished.
Segmentation fault: 11
```

Also tried removing the bundles folder but it doesn't help.

**Comment by SimpleX91** [comment]
Look here: https://github.com/junegunn/vim-plug/wiki/faq#vim-caught-deadly-signal-segv

**Comment by begriffs** [comment]
You're right, that's the problem! Could you add the following check to the script?

Run this `vim -c "ruby puts RUBY_VERSION" -c q` and check for nonzero exit code. The code is 139 on my system. If you detect the code then exit and print something like

```
The Ruby interface is broken on your installation of vim.
If you're on OS X, try the following

rvm use system
brew reinstall vim
```

After reinstalling vim on my system the check passes and your installer works! Vim-plug is super fast, and vim startup time feels faster too.

**Comment by SimpleX91** [comment]
Of course, I'll make it soon.
Try NeoVim, you'll be amazed ;)

**Comment by SimpleX91** [comment]
I've done it, but would that work ok for Vim compiled without Ruby support? It's not really required, see https://github.com/junegunn/vim-plug#pros
We need to check only for broken Ruby.

**Comment by begriffs** [comment]
So maybe the most comprehensive test is to first do `vim --version | grep +ruby` and if that returns 0 then check whether ruby is broken.

**Comment by begriffs** [comment]
I'll merge this and update the check.

**Pull request merged**